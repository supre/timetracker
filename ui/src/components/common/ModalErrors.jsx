import React from "react";
import { Modal, Message } from "semantic-ui-react";

export default ({ errors }) => {
  return errors.length > 0
    ? errors.map((error, key) => {
        return (
          <Modal.Content key={key + 1}>
            <Message error>{error}</Message>
          </Modal.Content>
        );
      })
    : "";
};
