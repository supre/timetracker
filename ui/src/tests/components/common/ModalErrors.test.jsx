import React from "react";
import renderer from "react-test-renderer";
import ModalErrors from "../../../components/common/ModalErrors";

test("Login", () => {
  // When there is atleast one error
  let login = renderer.create(<ModalErrors errors={["Test error"]} />);
  let tree = login.toJSON();
  expect(tree).toMatchSnapshot();

  // When there are no errors
  login = renderer.create(<ModalErrors errors={[]} />);
  tree = login.toJSON();
  expect(tree).toMatchSnapshot();
});
