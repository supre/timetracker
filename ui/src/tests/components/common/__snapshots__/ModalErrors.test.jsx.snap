// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login 1`] = `
<div
  className="content"
>
  <div
    className="ui error message"
  >
    Test error
  </div>
</div>
`;

exports[`Login 2`] = `""`;
